tests/streaming/test_dataset.py

```diff
@@ -85,10 +99,18 @@ def test_should_replace_path():
 
 
 @pytest.mark.parametrize("drop_last", [False, True])
-def test_streaming_dataset_distributed_no_shuffle(drop_last, tmpdir):
+@pytest.mark.parametrize(
+    "compression",
+    [
+        pytest.param(None),
+        pytest.param("zstd", marks=pytest.mark.skipif(condition=not _ZSTD_AVAILABLE, reason="Requires: ['zstd']")),
```

Comment:
But it is wired that if I apply the same compression param to the `test_streaming_dataset_distributed_full_shuffle_odd`, `test_streaming_dataset_distributed_full_shuffle_even` and `test_streaming_dataset_distributed_full_shuffle_even_multi_nodes` below, the test process will take a long time. And the test of `test_streaming_dataset_distributed_full_shuffle_odd`. And `test_streaming_dataset_distributed_full_shuffle_even` with zstd will cause the pytest session not to terminate even after all the test cases have been passed.

Reply:
And after some debug, I find that is the https://github.com/Lightning-AI/litdata/blob/a2413e98ad39328ecb84e0f7e16ae0523822ae87/tests/streaming/test_dataset.py#L228 which cause the pytest to hang. I am still debugging this.

Reply:
Interesting. Give it a try. If you are still blocked, we can pair debug this together.